src/components/Modal/Modal.tsx

```diff
@@ -0,0 +1,41 @@
+'use client';
+
+import { useEffect, useState, useRef } from 'react';
+import ReactDOM from 'react-dom';
+import classNames from 'classnames/bind';
+import { useOutsideClick } from '@/hooks/useOutsideClick';
+import styles from './Modal.module.scss';
+
+interface ModalPortalProps {
+  children: React.ReactNode;
+}
+
+interface ModalProps extends ModalPortalProps {
+  isOpen: boolean;
+  onClose: () => void;
+}
+
+function ModalPortal({ children }: ModalPortalProps) {
+  const [portalElement, setPortalElement] = useState<Element | null>(null);
+  useEffect(() => {
+    setPortalElement(document.getElementById('modal'));
+  }, []);
+  if (!portalElement) return null;
+  return ReactDOM.createPortal(children, portalElement) as JSX.Element;
+}
+
+export default function Modal({ children, isOpen, onClose }: ModalProps) {
```

Comment:
```suggestion
export default function Modal({ children, isOpen, onClose }: PropsWithChildren<ModalProps>) {
```
여기도 이렇게 바꿀 수 있습니다! (이렇게 쓴다면 위에 `extends ModalPortalProps` 지워도 될거 같아요!!)